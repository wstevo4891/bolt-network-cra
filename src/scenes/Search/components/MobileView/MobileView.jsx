import PropTypes from 'prop-types'
import React from 'react'

import SearchResults from '../SearchResults'

import Suggestions from '../Suggestions'

import SuggestionsLabel from '../SuggestionsLabel'

const MobileView = ({ handleClick, suggestion, query }) => (
  <>
    <div className="row">
      <div className="col-12 mb-2">
        <SuggestionsLabel>
          Results for &quot;{query}&quot;
        </SuggestionsLabel>
      </div>
    </div>

    <SearchResults suggestion={suggestion} />

    <div className="row">
      <Suggestions handleClick={handleClick} query={query} />
    </div>
  </>
)

MobileView.propTypes = {
  handleClick: PropTypes.func.isRequired,
  suggestion: PropTypes.object.isRequired,
  query: PropTypes.string.isRequired,
}

export default MobileView
