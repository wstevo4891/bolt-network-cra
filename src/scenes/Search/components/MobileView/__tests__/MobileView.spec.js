import React from "react";
import { shallow } from "enzyme";

import MobileView from "../MobileView";

const GENRES = [{ title: "Action" }, { title: "Sci-Fi" }];

const MOVIES = [
  { title: "The Avengers" },
  { title: "Alien" },
  { title: "Avatar" },
];

const PEOPLE = [
  { name: "Ridley Scott" },
  { name: "James Cameron" },
  { name: "Joss Whedon" },
];

describe("<MobileView />", () => {
  let wrapper, props;

  beforeEach(() => {
    props = {
      handleClick: jest.fn(),
      suggestion: {
        text: 'action',
        type: 'genre',
      },
      query: 'action',
    };
  });

  it("should match snapshot", () => {
    wrapper = shallow(<MobileView {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
