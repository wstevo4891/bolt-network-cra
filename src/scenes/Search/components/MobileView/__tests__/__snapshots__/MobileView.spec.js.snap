// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileView /> should match snapshot 1`] = `
<Fragment>
  <div
    className="row"
  >
    <div
      className="col-12 mb-2"
    >
      <SuggestionsLabel>
        Results for "
        action
        "
      </SuggestionsLabel>
    </div>
  </div>
  <SearchResults
    suggestion={
      Object {
        "text": "action",
        "type": "genre",
      }
    }
  />
  <div
    className="row"
  >
    <Suggestions
      handleClick={[MockFunction]}
      query="action"
    />
  </div>
</Fragment>
`;
